### ISSUE TITLE
arrow-stack does not specify version of "io.arrow-kt:arrow-optics-ksp-plugin"

### ISSUE THREAD

**Issue by MixusMinimax** [arrow-kt/arrow#3353]
### Arrow Version:
```
1.2.1
```

### My `gradle.build.kts`:
```kts
plugins {
    // ...
    id("com.google.devtools.ksp") version "1.9.21-1.0.15"
}
// ...
dependencies {
    // ...
    // Arrow
    implementation(platform("io.arrow-kt:arrow-stack:1.2.1"))
    implementation("io.arrow-kt:arrow-core")
    implementation("io.arrow-kt:arrow-fx-coroutines")
    implementation("io.arrow-kt:arrow-optics")
    ksp("io.arrow-kt:arrow-optics-ksp-plugin")
    implementation(kotlin("stdlib-jdk8"))
}
```

### Problem:

When not explicitly specifying the version of `"io.arrow-kt:arrow-optics-ksp-plugin"`, gradle can not find it.

```
> Task :example:kspKotlin FAILED

Execution failed for task ':example:kspKotlin'.
> Could not resolve all files for configuration ':example:kspKotlinProcessorClasspath'.
   > Could not find io.arrow-kt:arrow-optics-ksp-plugin:.
     Required by:
         project :example

Possible solution:
 - Declare repository providing the artifact, see the documentation at https://docs.gradle.org/current/userguide/declaring_repositories.html
```

### Reason:

`arrow-stack` constrains `"io.arrow-kt:arrow-optics-ksp-plugin"` as `api`, instead of `ksp`.

### Workarounds:

1. Specify version in build file manually
2. Add the block `constraints { ksp("io.arrow-kt:arrow-optics-ksp-plugin:1.2.1") }`
3. Use dependency libraries (toml file)

### Proposed Solution:

The implementation of `arrow-stack` should change:
```kts
api("io.arrow-kt:arrow-optics-ksp-plugin:$version")
```
to
```kts
// also add ksp to plugins? This might be a problem for people who do not use ksp.
ksp("io.arrow-kt:arrow-optics-ksp-plugin:$version")
```

### Question:

is ksp supposed to honor constraints marked as `api` and this is a ksp bug?

**Comment by MixusMinimax** [comment]
Update: This is the constraint function I am talking about:

```kt
/**
 * Adds a dependency constraint to the 'ksp' configuration.
 *
 * @param constraintNotation the dependency constraint notation
 *
 * @return the added dependency constraint
 *
 * @see [DependencyConstraintHandler.add]
 */
fun DependencyConstraintHandler.`ksp`(constraintNotation: Any): DependencyConstraint =
    add("ksp", constraintNotation)
```

As you can see, the function is very simple.

In order to add the constraint in `arrow-stack` without adding the ksp plugin, the following code can be used:
```kts
constraints {
    if (configurations.any { it.name == "ksp" }) {
        add("ksp", "io.arrow-kt:arrow-optics-ksp-plugin:1.2.1")
    }
}
```

**Comment by serras** [comment]
Fixed as of #3372, this will come out as part of 1.2.2
